### ISSUE TITLE
class_agnostic not working when training?

### ISSUE THREAD

**Issue by jinyu121** [jwyang/faster-rcnn.pytorch#259]
Hi~

I noticed this code https://github.com/jwyang/faster-rcnn.pytorch/blob/master/lib/model/faster_rcnn/faster_rcnn.py#L90

When using class_agnostic, the net will generate only one coordinate, that's right.
When not use it, the net will generate a coordinate for each class. BUT this code only select one of them, and send it to the loss function. Why not use them all, but only one? Any difference with `class_agnostic` method by doing this?

Can anyone explain the code and motivation? THX. 

----

Maybe I can not explain it clearly in English... Well, this is the native language that I want to say, you can translate it.....

我们只考虑训练的时候。
在开启class_agnostic的时候，对于每一个BoundingBox，将会只产生一组坐标——这是对的。
在不开启class_agnostic的时候，会对每一个类别产生一组坐标。但 https://github.com/jwyang/faster-rcnn.pytorch/blob/master/lib/model/faster_rcnn/faster_rcnn.py#L90 将这些坐标进行了选择，只选取了其中的一组坐标用来计算Loss。——那么，这样做就和开启class_agnostic没有区别了。为什么要这样做，而不是让所有坐标都参与Loss的计算？
有谁可以解释一下这样做的动机么？

**Comment by jwyang** [comment]
@jinyu121 in class-agnostic, the loss is only computed for the corresponding object classes, since the ground-truth is provided with the bounding box associated with the correct class label.
